common/ManagementCoreShared/SharedExtensions.cs

```diff
@@ -36,27 +36,79 @@ public static ResourceIdentifier AppendProviderResource(this ResourceIdentifier
         /// for example, 'subnets'. </param>
         /// <param name="childResourceName"> The name of the resource. </param>
         /// <returns> The combined resource id. </returns>
-        [EditorBrowsable(EditorBrowsableState.Never)]
         public static ResourceIdentifier AppendChildResource(this ResourceIdentifier identifier, string childResourceType, string childResourceName)
         {
             ValidateChildResourceParameters(childResourceType, childResourceName);
             return new ResourceIdentifier(identifier, childResourceType, childResourceName);
         }
 
-        internal static void ValidateProviderResourceParameters(string providerNamespace, string resourceType, string resourceName)
+        /// <summary>
+        /// An extension method for supporting replacing one dictionary content with another one.
+        /// This is used to support resource tags.
+        /// </summary>
+        /// <param name="dest"> The destination dictionary in which the content will be replaced. </param>
+        /// <param name="src"> The source dictionary from which the content is copied from. </param>
+        /// <returns> The destination dictionary that has been altered. </returns>
+        public static IDictionary<string, string> ReplaceWith(this IDictionary<string, string> dest, IDictionary<string, string> src)
+        {
+            dest.Clear();
+            foreach (var kv in src)
+            {
+                dest.Add(kv.Key, kv.Value);
```

Comment:
I think we can use this directly.
```suggestion
                dest.Add(kv);
```